### ISSUE TITLE
Not deterministic behavior in patch tree append

### ISSUE THREAD

**Issue by priv-kweihmann** [phytec/meta-phytec#11]
https://github.com/phytec/meta-phytec/blob/ea8941a818e28b2be16dd57e184ba7814bb0c438/classes/patch-auto-append.bbclass#L25 uses os.listdir, which is in the end depending on the inode order of the underlying fs. Therefore you could have different results on different hosts (or even on the same host at different points in time)

Either sort the list using `sorted(os.listdir(...))` or find another alternative to make the function reproducible.

And I think this function needs some further treatment... here are my suggestions

- remove the print call or replace it with bb.debug or bb.note
- rework the whole block of

```
for filename in (f for f in os.listdir(dir)):
    path = os.path.join(dir, filename)
    print(path, filename)

    if os.path.isfile(path) and filename.endswith(".patch"):
        # TODO Escape evil characters ;-)
        src_uri_patches.append("file://" + filename)
 ```
 
 to 
 
 ```
 for filename in sorted(glob.glob(os.path.join(dir, "*.patch"))):
    if os.os.path.isfile(filename):
        src_uri_patches.append("file://%s" % os.path.relpath(d.getVar("WORKSPACE"), filename))
 ```

**Comment by smk-embedded** [comment]
Thanks for the hint. I will have look at it.

**Comment by smk-embedded** [comment]
fixed in commit 2cfb5b68189837df744b41e62d16e13b7883bad0
